src/applicationManager.ts

```diff
@@ -528,3 +546,228 @@ async function getGraphAPIRoles(token: string, graphApiPermissions: string[]) {
   }
   return { appRoleIds, graphAPIObjectId };
 }
+
+/**
+ * Updates an Azure Entra Application, specifically with a collection of new App Roles
+ * @param applicationId The object ID of application registration
+ * @param appRoles A collection (array) of AppRole custom objects to update the application with
+ */
+export async function updateApplicationAppRoles({
+  token,
+  applicationId,
+  appRoles,
+}: {
+  token: string;
+  applicationId: string;
+  appRoles: Array<AppRole>;
+}): Promise<void> {
+  let body = {
+    appRoles: appRoles,
+  };
+  const result = await fetch(
+    `https://graph.microsoft.com/v1.0/applications/${applicationId}`,
+    {
+      method: "PATCH",
+      headers: {
+        Authorization: `Bearer ${token}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(body),
+    },
+  );
+  await errorHandler("updating application config", result);
+}
+
+/**
+ * Create formatted AppRole object which is expected by graph api
```

Comment:
seems like this function could be unneeded you could just inject the extra properties in when you send it pretty easily with spread operator to merge two object.

Reply:
Ah nice - I was struggling with this because of `groups` existing in the array but not being needed yet - good old genAI suggested to destructure 😁 

https://github.com/hmcts/azure-app-proxy-manager/pull/122/commits/64d3cf7fddf24cb820f67958486b8fb32cb5e23d